rules/rbe_repo/container.bzl

```diff
@@ -239,17 +303,32 @@ def run_and_extract(
     # Set the Bazel version that Bazelisk will use
     docker_run_flags += ["--env", ("USE_BAZEL_VERSION=%s" % bazel_version_string)]
 
+    # Run as the root user on Linux and equivalent ContainerAdministrator on Windows for images
+    # that have a default USER set in the Dockerfile
+    if _is_windows(ctx):
+        docker_run_flags += ["--user", "ContainerAdministrator"]
+    else:
+        docker_run_flags += ["--user", "root"]
+
     # Command to copy resources used for rbe_autoconfig to the container.
     copy_data_cmd = []
 
     # Command to clean up the data volume container.
     clean_data_volume_cmd = ""
-    copy_data_cmd.append("data_volume=$(docker create -v " + _ROOT_DIR + " " + image_name + ")")
-    copy_data_cmd.append(docker_tool_path + " cp $(realpath " + project_root + ") $data_volume:" + _REPO_DIR)
-    copy_data_cmd.append(docker_tool_path + " cp " + str(ctx.path("container")) + " $data_volume:" + _ROOT_DIR + "/container")
-    copy_data_cmd.append(docker_tool_path + " cp " + str(ctx.path("bazelisk")) + " $data_volume:" + _BAZELISK_PATH)
-    docker_run_flags += ["--volumes-from", "$data_volume"]
-    clean_data_volume_cmd = docker_tool_path + " rm $data_volume"
+
+    # docker cp does not function on Windows as expected when copying into volumes so we use bind
+    # mounts instead
+    if _is_windows(ctx):
```

Comment:
We had to do the following as `docker cp` and volumes don't seem to work as expected

See: https://gist.github.com/sunjayBhatia/d235b8ff226fdee4025df8da34f4f5c5 for a repro example

Reply:
Ack